client/python/openlineage/client/client.py

```diff
@@ -27,55 +25,48 @@ class OpenLineageClientOptions:
 
 class OpenLineageClient:
     def __init__(
-            self,
-            url: str,
-            options: OpenLineageClientOptions = OpenLineageClientOptions(),
-            session: Session = None
+        self,
+        url: Optional[str] = None,
+        options: Optional[OpenLineageClientOptions] = None,
+        session: Optional[Session] = None,
+        transport: Optional[Transport] = None,
     ):
-        url = url.strip()
-        try:
-            parsed = parse_url(url)
-            if not (parsed.scheme and parsed.netloc):
-                raise ValueError(f"Need valid url for OpenLineageClient, passed {url}")
-        except Exception as e:
-            raise ValueError(f"Need valid url for OpenLineageClient, passed {url}. Exception: {e}")
-        self.url = url
-        self.options = options
-        self.session = session if session else Session()
-        self.session.headers['Content-Type'] = 'application/json'
+        if url:
+            # Backwards compatibility: if URL is set, use old path to initialize
+            # HTTP transport.
+            if not options:
+                options = OpenLineageClientOptions()
+            if not session:
+                session = Session()
+            self._initialize_url(url, options, session)
+        elif transport:
+            self.transport = transport
+        else:
+            self.transport = get_default_factory().create()
 
-        if self.options.api_key:
-            self._add_auth(options.api_key)
-        if self.options.adapter:
-            self.session.mount(self.url, options.adapter)
+    def _initialize_url(
+        self,
+        url: str,
+        options: OpenLineageClientOptions,
+        session: Session
+    ):
+        self.transport = HttpTransport(HttpConfig(
+            url=url,
+            timeout=options.timeout,
+            verify=options.verify,
+            api_key=options.api_key,
+            session=session,
+            adapter=options.adapter
```

Comment:
I'd leave the options parsing to the `HttpConfig`, rather than passing specific keys here. Allow the `HttpConfig` to support new features without having to change the client code here.

Reply:
In the `openlineage-java` lib, we take the approach you're suggestion here @collado-mike. That is, we instantiate the [`HttpConfig`](https://github.com/OpenLineage/OpenLineage/blob/feature/http-client-java/client/java/src/main/java/io/openlineage/client/transports/HttpConfig.java) object based on what transport is enabled in [`openlineage.yml`](https://github.com/OpenLineage/OpenLineage/tree/feature/http-client-java/client/java#configuration) file